instance/tests/management/test_kill_zombies.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2019 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Tests for the 'kill_zombies' management command.
+"""
+# Imports #####################################################################
+
+from unittest.mock import patch
+from django.core import mail
+from django.core.management import CommandError
+from django.test import TestCase, override_settings
+from instance.tasks import KillZombiesRunner, kill_zombies_periodically
+
+# Tests #######################################################################
+
+
+@patch.dict(
+    "os.environ",
+    {
+        "KILL_ZOMBIES_ENABLED": "true",
+        "KILL_ZOMBIES_REGION": "some-region",
+    },
+)
+class KillZombiesPeriodicallyTestCase(TestCase):
+    """
+    Test cases for the `kill_zombies_periodically` periodic task.
+    """
+    # pylint: disable= bad-continuation
+    @patch.dict(
+        "os.environ",
+        {
+            "KILL_ZOMBIES_ENABLED": "",
+        },
+    )
+    @patch("instance.tasks.call_command")
+    def test_toggle_task(self, mock_command):
+        """
+        Tests that the task doesn't run unless the env variable
+        KILL_ZOMBIES_ENABLED is set and true
+        """
+        kill_zombies_periodically()
+        self.assertFalse(mock_command.called)
+
+        with patch.dict("os.environ", {"KILL_ZOMBIES_ENABLED": "true"}):
+            kill_zombies_periodically()
+            self.assertTrue(mock_command.called)
+
+    @patch("instance.tasks.call_command")
+    def test_set_env_variables(self, mock_command):
+        """
+        Tests that the task passes env variables to
+        the kill_zombies command call
+        """
+        kill_zombies_periodically()
+        mock_command.assert_called_with(
+            "kill_zombies", region="some-region"
+        )
+
+    @patch("instance.management.commands.kill_zombies.openstack_utils.get_nova_client")
```

Comment:
@spokerman12, as discussed already, the `kill_zombies` command should be treated as a blackbox that works and hence there is no need to test it or anything that it invokes.

Reply:
Alright. Thanks for clarifying this. Modified to fit requirements.